app/bundles/CampaignBundle/Membership/MembershipBuilder.php

```diff
@@ -154,7 +159,8 @@ private function addNewlyQualifiedMembers(int $totalContactsProcessed): int
             $this->manager->addContacts($contactCollection, $this->campaign, false);
 
             // Clear Lead entities from RAM
-            $this->leadRepository->clear();
+            $this->objectManager->flush();
+            $this->objectManager->clear();
```

Comment:
```suggestion
            $this->leadRepository->detachEntities(contacts);
```
Let's replace it with detaching entities.